codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestDecoder.java

```diff
@@ -121,32 +133,73 @@ public HttpRequestDecoder(
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED);
     }
 
+    /**
+     * @deprecated Prefer the {@link #HttpRequestDecoder(int, int, int)} constructor,
+     * to always have header validation enabled.
+     */
+    @Deprecated
     public HttpRequestDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders) {
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, validateHeaders);
     }
 
+    /**
+     * @deprecated Prefer the {@link #HttpRequestDecoder(int, int, int, int)} constructor,
+     * to always have header validation enabled.
+     */
+    @Deprecated
     public HttpRequestDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders,
             int initialBufferSize) {
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, validateHeaders,
               initialBufferSize);
     }
 
+    public HttpRequestDecoder(
+            int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, int initialBufferSize) {
+        super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, DEFAULT_VALIDATE_HEADERS,
+              initialBufferSize);
+    }
+
+    /**
+     * @deprecated Prefer the {@link #HttpRequestDecoder(int, int, int, int, boolean)} constructor,
+     * to always have header validation enabled.
+     */
+    @Deprecated
     public HttpRequestDecoder(
             int maxInitialLineLength, int maxHeaderSize, int maxChunkSize, boolean validateHeaders,
             int initialBufferSize, boolean allowDuplicateContentLengths) {
         super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, validateHeaders,
               initialBufferSize, allowDuplicateContentLengths);
     }
 
+    public HttpRequestDecoder(
+            int maxInitialLineLength, int maxHeaderSize, int maxChunkSize,
+            int initialBufferSize, boolean allowDuplicateContentLengths) {
+        super(maxInitialLineLength, maxHeaderSize, maxChunkSize, DEFAULT_CHUNKED_SUPPORTED, DEFAULT_VALIDATE_HEADERS,
+              initialBufferSize, allowDuplicateContentLengths);
+    }
+
+    /**
+     * @deprecated Prefer the {@link #HttpRequestDecoder(int, int, int, int, boolean, boolean)} constructor,
+     * to always have header validation enabled.
+     */
+    @Deprecated
```

Comment:
If people opt in to warnings-as-errors, then it's because they want a very strong signal when new warnings show up. We need to be able to deprecate things, as our APIs and recommendations evolve.

We can't reduce the visibility of existing methods. That would be an actual breaking change.

Reply:
Agreed, no breaking changes should be made. I was thinking of changing the order or parameters if we go with `protected` option. Let's see if we can explore a "function for `HttpHeaders`" approach first.